src/python/gudhi/datasets/remote.py

```diff
@@ -204,20 +242,91 @@ def fetch_bunny(file_path = None, accept_license = False):
     """
 
     file_url = "https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/bunny/bunny.npy"
-    file_checksum = 'f382482fd89df8d6444152dc8fd454444fe597581b193fd139725a85af4a6c6e'
+    file_checksum = "f382482fd89df8d6444152dc8fd454444fe597581b193fd139725a85af4a6c6e"
     license_url = "https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/bunny/bunny.LICENSE"
-    license_checksum = 'b763dbe1b2fc6015d05cbf7bcc686412a2eb100a1f2220296e3b4a644c69633a'
+    license_checksum = "b763dbe1b2fc6015d05cbf7bcc686412a2eb100a1f2220296e3b4a644c69633a"
 
     archive_path = _get_archive_path(file_path, "points/bunny/bunny.npy")
 
     if not exists(archive_path):
         _fetch_remote(file_url, archive_path, file_checksum)
         license_path = join(split(archive_path)[0], "bunny.LICENSE")
-        _fetch_remote(license_url, license_path, license_checksum)
-        # Print license terms unless accept_license is set to True
-        if not accept_license:
-            if exists(license_path):
-                with open(license_path) as f:
-                    print(f.read())
-
-    return np.load(archive_path, mmap_mode='r')
+        _fetch_remote_license(license_url, license_path, license_checksum, accept_license)
+
+    return np.load(archive_path, mmap_mode="r")
+
+
+@cache
+def _load_and_cache_activity(file_path):
+    return np.load(file_path)
```

Comment:
I would suggest either `mmap_mode="r"` or doing `X.flags.writeable=False` after the load, to prevent surprises.